src/MainWindow.vala

```diff
@@ -180,154 +155,146 @@ public class Mail.MainWindow : Hdy.ApplicationWindow {
-#if HAS_SOUP_3
-            var hover_url = url != null ? GLib.Uri.unescape_string (url) : null;
-#else
-            var hover_url = url != null ? Soup.URI.decode (url) : null;
-#endif
-
-            if (hover_url == null) {
-                message_overlay.hide ();
-            } else {
-                message_overlay.label = hover_url;
-                message_overlay.show ();
-            }
-        });
+       //  var scrolled_child = message_list_scrolled.get_child ();
+       //  if (scrolled_child is Gtk.Box) {
+       //      ((Gtk.Box) scrolled_child).set_focus_vadjustment (new Gtk.Adjustment (0, 0, 0, 0, 0, 0));
+       //  }
+
+       //  view_overlay = new Gtk.Overlay ();
+       //  view_overlay.add_overlay (message_list_scrolled);
+
+       //  var message_list_container = new Gtk.Grid ();
+       //  message_list_container.add_css_class (Granite.STYLE_CLASS_BACKGROUND);
+       //  message_list_container.attach (headerbar, 0, 0);
+       //  message_list_container.attach (view_overlay, 0, 1);
+
+       //  var message_overlay = new Granite.OverlayBar (view_overlay);
+       //  message_overlay.no_present = true;
+
+       // message_list_box.hovering_over_link.connect ((label, url) => {
+       //      var hover_url = url != null ? GLib.Uri.unescape_string (url) : null;
+       //      if (hover_url == null) {
+       //          message_overlay.hide ();
+       //      } else {
+       //          message_overlay.label = hover_url;
+       //          message_overlay.show ();
+       //      }
+       //  });
 
         paned_start = new Gtk.Paned (Gtk.Orientation.HORIZONTAL);
-        paned_start.pack1 (folders_list_view, false, false);
-        paned_start.pack2 (conversation_list_grid, true, false);
+        paned_start.set_start_child (folders_list_view);
+        paned_start.set_end_child (conversation_list_grid);
 
         paned_end = new Gtk.Paned (Gtk.Orientation.HORIZONTAL);
-        paned_end.pack1 (paned_start, false, false);
-        paned_end.pack2 (message_list_container, true, true);
+        paned_end.set_start_child (paned_start);
+        // paned_end.set_start_child (message_list_container);
 
-        var welcome_view = new Mail.WelcomeView ();
+        // var welcome_view = new Mail.WelcomeView ();
 
         var placeholder_stack = new Gtk.Stack ();
         placeholder_stack.transition_type = Gtk.StackTransitionType.OVER_DOWN_UP;
         placeholder_stack.add_named (paned_end, "mail");
-        placeholder_stack.add_named (welcome_view, "welcome");
+        //placeholder_stack.add_named (welcome_view, "welcome");
 
-        add (placeholder_stack);
+        set_content (placeholder_stack);
 
-        var header_group = new Hdy.HeaderGroup ();
-        header_group.add_header_bar (folders_list_view.header_bar);
-        header_group.add_header_bar (search_header);
-        header_group.add_header_bar (headerbar);
+        //@TODO: lookup a new implementation for Hdy.HeaderGroup
```

Comment:
The way to do this in Gtk4 is to use [Gtk.WindowControls](https://valadoc.org/gtk4/Gtk.WindowControls.html) instead and you can set the `side` property